src/components/Subthoughts.js

```diff
@@ -286,24 +306,12 @@ export const Subthoughts = connect(({ cursorBeforeEdit, cursor, contextViews, th
             ['distance-from-cursor-' + distance]: true
           })}
         >
-          {children
-            .filter(child => {
-              const value = showContexts ? head(child.context) : child.value
-              return showHiddenThoughts ||
-                (!isFunction(value) && !meta(pathToContext(unroot(thoughtsRanked)).concat(value)).hidden)
-            })
+          {filteredChildren
             .map((child, i) => {
-
-              const value = showContexts ? head(child.context) : child.value
-
-              // Because the current thought only needs to hash match another thought, we need to use the exact value of the child from the other context
-              // child.context SHOULD always be defined when showContexts is true
```

Comment:
Let's preserve this comment in `getChildPath` as it explains some important invariants for this function.

Reply:
ok